sky/spot/recovery_strategy.py

```diff
@@ -142,39 +156,66 @@ def _launch(self, max_retry=3, raise_on_failure=True) -> Optional[float]:
                 # code.
                 logger.info(
                     f'Failed to launch the spot cluster with error: {e}')
-
-            cluster_status, _ = backend_utils.refresh_cluster_status_handle(
-                self.cluster_name, force_refresh=True)
-            if cluster_status == global_user_state.ClusterStatus.UP:
-                # Wait the job to be started
+                retry_launch = True
+                exception = e
+
+            status = None
+            job_checking_retry_cnt = 0
+            # Check the job status until it is not in initialized status.
+            while True:
```

Comment:
`while not retry_launch:`? So that if we hit L159, this loop needs not be entered.

Reply:
Good catch! Fixed. Thanks!